src/mappings/genesis.ts

```diff
@@ -0,0 +1,443 @@
+import {
+  Event,
+  GenesisFile as GenesisEntity,
+} from "../types";
+import type { AccountProps } from "../types/models/Account";
+import type { ApplicationProps } from "../types/models/Application";
+import { ApplicationGatewayProps } from "../types/models/ApplicationGateway";
+import type { ApplicationServiceProps } from "../types/models/ApplicationService";
+import type { AppParamProps } from "../types/models/AppParam";
+import type { BalanceProps } from "../types/models/Balance";
+import type { GatewayProps } from "../types/models/Gateway";
+import type { GenesisBalanceProps } from "../types/models/GenesisBalance";
+import { MsgAddServiceProps } from "../types/models/MsgAddService";
+import { MsgDelegateToGatewayProps } from "../types/models/MsgDelegateToGateway";
+import { MsgStakeApplicationProps } from "../types/models/MsgStakeApplication";
+import { MsgStakeApplicationServiceProps } from "../types/models/MsgStakeApplicationService";
+import { MsgStakeGatewayProps } from "../types/models/MsgStakeGateway";
+import { MsgStakeSupplierProps } from "../types/models/MsgStakeSupplier";
+import { MsgStakeSupplierServiceProps } from "../types/models/MsgStakeSupplierService";
+import type { NativeBalanceChangeProps } from "../types/models/NativeBalanceChange";
+import type { ServiceProps } from "../types/models/Service";
+import type { SupplierProps } from "../types/models/Supplier";
+import { SupplierServiceConfigProps } from "../types/models/SupplierServiceConfig";
+import type { TransactionProps } from "../types/models/Transaction";
+import { configOptionsFromJSON, rPCTypeFromJSON } from "../types/proto-interfaces/poktroll/shared/service";
+import { StakeStatus, TxStatus } from "./constants";
+import type { Genesis } from "./types/genesis";
+import {
+  getAppDelegatedToGatewayId,
+  getBalanceId,
+  getGenesisFakeTxHash,
+  getMsgStakeServiceId, getParamId,
+  getStakeServiceId,
+  stringify
+} from "./utils";
+
+export async function handleGenesis(block: CosmosBlock): Promise<void> {
+  const genesis: Genesis = require('../../genesis.json');
+
+  // IMPORTANT: Return early if this is not the genesis initial height as this is called for block indexed!
+  if (block.block.header.height !== genesis.initial_height) {
+    return
+  }
+
+  logger.info(`[handleGenesis] (block.header.height): indexing genesis block ${block.block.header.height}`);
+
+  // save records
+  await Promise.all([
+    _handleGenesisEvent(block),
+    _handleGenesisServices(genesis, block),
+    _handleGenesisBalances(genesis, block),
+    _handleGenesisGateways(genesis, block),
+    _handleGenesisSuppliers(genesis, block),
+    _handleGenesisApplications(genesis, block),
+    // PARAMS
+    _handleGenesisParams(genesis, block),
+  ]);
+
+  await GenesisEntity.create({
+    id: block.block.header.height.toString(),
+    raw: JSON.stringify(genesis),
+  }).save();
+}
+
+async function _handleGenesisEvent(block: CosmosBlock): Promise<void> {
```

Comment:
#PUC Why do we need a genesis event? Is it just to be able to query for the genesis height?

Reply:
We just added it to have an event for the entities that requires it. Doing this we are still requiring the events in the schema.